codegen/templates/tchannel_client.tmpl

```diff
@@ -180,7 +194,35 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 	}
 }
 
-func initializeDynamicChannel(deps *module.Dependencies, headerPatterns []string, altChannelMap map[string]*tchannel.SubChannel, re ruleengine.RuleEngine) ([]string, ruleengine.RuleEngine) {
+func createNewTchannelForClient(deps *module.Dependencies, serviceName string) *tchannel.Channel {
```

Comment:
this doesn't depend on the template so let's move this out so you don't have to generated this several times during the code-gen step.

Reply:
probably someplace here: https://github.com/uber/zanzibar/blob/master/runtime/tchannel_client.go

Reply:
Will lead to an explosion of parameters in the `tchannel_client.go` as it needs a tracer, logger etc.

I found a better place for it in `gateway.go` which will also clean up the `defaultDependencies struct` a bit and would avoid repeated codegen. Thanks for the input.